### PR TITLE
Simultaneous playback capture shellcheck fixes

### PR DESCRIPTION
**Pull request by greg-intel**
Fix shellcheck errors per https://github.com/thesofproject/sof-test/issues/729

Corrected a source reference. Fixed various quoting issues.  Corrected exit codes and updated syntax.

Signed-off-by: Greg Galloway <greg.galloway@intel.com>

### CODE REVIEW

**Review by marc-hb — APPROVED** [review]
Code changes look good, thanks!

Nit: this commit message seems inaccurate "The previous source reference for lib.sh triggers shellcheck errors."

**Comment by greg-intel** [comment]
> The previous source reference for lib.sh triggers shellcheck errors.

I do like this phrasing much more.  If I adjust the source reference in future files, I'll call this out properly.

**Comment by marc-hb** [comment]
> Nit: this commit message seems inaccurate "The previous source reference for lib.sh triggers shellcheck errors."

I thought this was referring to the addition of the meta `# shellcheck source=` comment.

Now I realize this is probably just a confusing way to say "add missing quotes"?

File: test-case/simultaneous-playback-capture.sh [hunk 1]
```diff
@@ -74,38 +75,38 @@ do
     setup_kernel_check_point
     dlogi "===== Testing: (Loop: $i/$loop_cnt) ====="
     # following sof-tplgreader, split 'both' pipelines into separate playback & capture pipelines, with playback occurring first
-    for order in $(seq 0 2 $(expr $PIPELINE_COUNT - 1))
+    for order in $(seq 0 2 $(( "$PIPELINE_COUNT" - 1)))
```
Comment (marc-hb) [inline test-case/simultaneous-playback-capture.sh:78]:
```suggestion
    for order in $(seq 0 2 $(( PIPELINE_COUNT - 1)))
```

**Comment by marc-hb** [comment]
All tests passed, merging

**Pull request merged**